priv/repo/migrations/20221007120337_create_job_posts.exs

```diff
@@ -0,0 +1,42 @@
+defmodule Erlef.Repo.Migrations.CreateJobPosts do
+  use Ecto.Migration
+
+  def change do
+    create table(:job_posts) do
+      add :title, :string, null: false
+      add :description, :text, null: false
+      add :position_type, :string, null: false
+      add :approved, :boolean, null: false, default: false
+      add :city, :string
+      add :region, :string
+      add :country, :string
+      add :postal_code, :string
+      add :remote, :boolean, null: false
+      # There's no standard limit on URI length so `text` is more appropriate
+      # than `string`.
+      add :website, :text
+      add :ttl, :interval, null: false, default: "30 days"
+      add :created_by, references(:members), null: false
+
+      timestamps()
+    end
+
+    execute(&create_expired_at_column/0, &drop_expired_at_column/0)
+
+    create index(:job_posts, [:created_by])
+  end
+
+  defp create_expired_at_column() do
+    query = """
+    ALTER TABLE job_posts
```

Comment:
This is pretty cool and a TIL. Only concern is putting logic in the database. Some times that's the right tool for the job, but is there a reason why we would prefer this over doing this in the app when we insert or update, performance cost aside?

Reply:
Everything in the database is more or less related to logic. Since the expiration date is a derived state, I think it's best to leave the owner of the data manage it. In this case it simplifies the application code because there's no need to think how is the record inserted. Is it inserted by `Repo.insert` or a raw query. Also Postgres has very nice functions and operators for working with intervals.

Reply:
Postgres has very nice functions indeed 😁 However, expired_at at least to me is application logic (I definitely should have said application logic), which doesn't belong in the database. I can always play devil's advocate and argue why this might be better done by the databases. However,  and for arguments sake if this syntax changes, becomes deprecated, etc. a migration in the database must be applied, what's more this feature requires upgrading the database to support it. If it was just a small operation in the application as part of the changeset, it would always work 😁  

I provided the above merely for communication purposes (i.e., so I'm more clear about my rationale).  That said, I'm not going to block on this one, as my feelings are not that strong and we do need upgrade the database 😄 

On a related note, it looks like the a test is failing because of this change. Haven't dug into why other than I saw a syntax error. Wrong version of postgres used in workflow perhaps?